rel/expr_binary.go

```diff
@@ -306,7 +307,28 @@ func evalValForAddArrow(lhs, rhs Value) (Value, error) {
 		}
 	}
 
+	if lhs, ok := lhs.(Dict); ok {
+		if rhs, ok := rhs.(Dict); ok {
+			return mergeDicts(lhs, rhs), nil
+		}
+	}
+
 	return nil, errors.Errorf(
-		"Both args to %q must be tuples, not %T and %T",
+		"Both args to %q must be tuples or dicts, not %T and %T",
```

Comment:
Nit: presumably you can't merge a dict into a tuple
```suggestion
		"Both args to %q must be both tuples or both dicts, not %T and %T",
```